src/v/cluster/partition.h

```diff
@@ -135,7 +135,7 @@ class partition {
         return _raft->config();
     }
 
-    partition_probe& probe() { return _probe; }
+    partition_probe probe() { return partition_probe(_probe); }
```

Comment:
> Nit: If `_probe` is a `partition_probe`, then this can still return by reference (although value is also fine, it's just that shared ownership isn't required for lifetime management, it's just enabled by the implementation). That would reduce coupling to the impl.

At that point, `partition_probe` could be a `std_unique_ptr<impl>` (if impl has a virtual destructor).

Reply:
agreed. if we don't need reference counting lets use a unique pointer.